lib/middleware/EnviosBD.dart

```diff
@@ -0,0 +1,209 @@
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+import 'package:transportify/middleware/Datos.dart';
+import 'package:transportify/modelos/PuntoTransportify.dart';
+
+class EnviosBD {
+  static const String coleccion_envios = 'envios';
+
+  static const String atributo_estado = 'estado';
+  static const String atributo_id_paquete = 'id_paquete';
+  static const String atributo_id_viaje = 'id_viaje';
+
+  static String obtenerEstado(DocumentSnapshot snapshot) => snapshot[atributo_estado];
+  static String obtenerIdPaquete(DocumentSnapshot snapshot) => snapshot[atributo_id_paquete];
+  static String obtenerIdViaje(DocumentSnapshot snapshot) => snapshot[atributo_id_viaje];
+
+
+static Function(BuildContext, AsyncSnapshot<QuerySnapshot>) _obtenerListaEnviosBuilder(Function(int estado) onTapMethod) {
+    Function(BuildContext, AsyncSnapshot<dynamic>) listaEnviosBuilder =  (BuildContext context, AsyncSnapshot<dynamic> snapshot) {
+      //Metodo constructor del dropdown
```

Comment:
Necesitas comprobar siempre si el snapshot tiene datos antes de hacer nada con ellos, para que si aún no están listos muestre de alguna manera que está cargando (de hecho si no lo haces, te salta un error).

```suggestion
      if (!snapshot.hasData) return const Text('Cargando...');
      
      //Metodo constructor del dropdown
```